app/models/archiver.rb

```diff
@@ -38,10 +38,11 @@ def self.archive_previous_versions
   def self.create_archives(source, archive, previous_version, production_version)
     return if archive.blank?
 
-    str = <<~SQL
-      INSERT INTO #{archive.table_name}
-      SELECT s.* FROM #{source.table_name} s
-    SQL
+    cols = archive.column_names
+    insert_cols = (cols.map { |col| '"' + col + '"' }).join(', ')
+    select_cols = (cols.map { |col| 's.' + col }).join(', ')
+
+    str = "INSERT INTO #{archive.table_name}(#{insert_cols}) SELECT #{select_cols} FROM #{source.table_name} s "
```

Comment:
I'm guessing this extra "s" was from a save mistake OR to pluralize something, but wanted to flag this as a possible typo

Reply:
Oh maybe not actually, I see the "s" on line 43

Reply:
It's not a typo, it's an alias for a table